src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,102 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.ArrayList;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(UserServlet.class.getName());
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // returns a list of events
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    Gson gson = new Gson();
+    response.setContentType("application/json");
+
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      Key userKey = KeyFactory.createKey("User", userEmail);
+      try {
+        Entity userEntity = datastore.get(userKey);
+        List<Entity> results = new ArrayList<>();
+        if(request.getParameter("get")!= null && request.getParameter("get").equals("saved")) {
```

Comment:
Consider use switch case here like
```
switch (request.getParameter("get")) {
  case "saved": handleSaved(); break;
  case "created": handleCreated(); break;
  default: // throw exception
}
private void handleSaved() {
....
```
This pattern will be easier to expand in the future.

Reply:
Makes sense, thanks!